Kestrel edits for lmod (#492)
NREL/HPC: docs/Documentation/Environment/lmod.md

# Environment Modules on NREL HPC

Kestrel and Eagle uses the *Lmod* environment modules system to easily manage software environments. 
Modules facilitate the use of different versions of applications, libraries, and toolchains, which enables support of multiple package versions concurrently.

Modules typically just set environment variables that one might traditionally do manually by, for example, adding `export` or `setenv` commands to their login script. 
Modules add the ability to back out changes in an orderly manner as well, so users can change their environment in a reversible way.

Modulefiles can be hosted in any location that the user makes known to the Lmod system via the module use command. 
By copying the reference modules, we set up into custom locations and modifying to your own tastes, you can host personal, project, or multi-project collections that you have complete control over.

> Our production location for module examples is `/nopt/nrel/apps/modules/default` on Kestrel and `/nopt/nrel/apps/modules/default` on Eagle. 

Reviewer: Instead of listing the paths  is there a way to explain to users how to see the production location on whichever system they are on?

This path is set for everyone by default. 

> In addition, motivated users who would like to support an application for the *"energy community"* on Eagle may request write access to the *"e-com"* space. 

Reviewer: I think could just be left on the Eagle "Modules" page since it only applies to Eagle.

|`avail`	                |Prints available modules grouped by path. |
|`list`	                    |Prints all currently loaded modules.|
|`display 'name'`	        |Prints settings and paths specified for a particular module.|
|`help 'name'`	            |Prints help message for a particular module.|
|`load 'name'`	            |Loads particular module. For modules listed as the '(default)', the short package name is sufficient. To load another version of the package the long package name is required (e.g., `module load fftw/3.3.10-cray-mpich-intel`).|
|`unload 'name'`	        |Unloads particular module.|
|`swap 'name1' 'name2'`	|First unload `modName1` and then load `modName2`.|
|`use {-a} A_PATH`	    |Prefix `{suffix}` the path `$A_PATH` to your `$MODULEPATH` variable, in order to find modules in that location.|
|`unuse A_PATH`	        |Remove the path `$A_PATH` from your `$MODULEPATH` variable.|

## Module Organization on NREL HPC 

The modulefiles that we provide are only a starting point. 
For maximum control, users should copy these files from the locations in `/nopt` to their own locations for which they have write access.

> Module files for baseline *applications*, *libraries*, *frameworks*, and *toolchains* are located in: 

Reviewer: Could add Swift and Vermilion

|HPC | Modulefies Location |
|:---:|:-----|
|Kestrel | `/nopt/nrel/apps/modules/default ` |
|Eagle | `/nopt/nrel/apps/modules/default/modulefiles/ ` |

Note: 
It should be mentioned that for Kestrel the modulefiles are organized under 3 categories

1-  Compilers and Mpis 
2-  Utilities and Libraries 
3-  Applications 

Users may and should freely copy these example modulefiles to preferred locations and customize them for their own use cases. 

> This can be particularly desirable to preserve a critical workflow as the software environment changes on Kestrel and Eagle, or to change the behavior, e.g., turn off automatic loading of prerequisites. 

Reviewer: ```suggestion
This can be particularly desirable to preserve a critical workflow as the software environment is not the same between systems, or to change the behavior, e.g., turn off automatic loading of prerequisites. 
```
Or something to similar to make it general
